### DISCUSSION TITLE
Differences between Python and C++ implementation

### DISCUSSION THREAD

**Discussion by hashb** [hungpham2511/toppra#211] [Q&A] [answered]
I recently switched to using C++ implementation of toppra and have been running into failed parameterization due to `ERR_UNKNOWN` and `ERR_FAIL_FORWARD_PASS`. Another thing that I noticed is that, for the same trajectory, the time to compute parameterization is around 0.04s for python but is taking around 0.1-0.2s in c++ which is counter intuitive for me. Is there any major differences between the python and C++ implementations? I am planning to dig deeper into this and wanted to see if I can get some insights from you @hungpham2511

**Comment by hungpham2511** [comment, marked as answer]
> Another thing that I noticed is that, for the same trajectory, the time to compute parameterization is around 0.04s for python but is taking around 0.1-0.2s in c++ 

Could you provide more information on the solver used?

If you use Seidel solver in C++ you should have similar timings.

>  ERR_UNKNOWN and ERR_FAIL_FORWARD_PASS

It would be helpful if you can provide a failing example.

**Comment by hashb** [comment]
Thank you for the quick reply! I was using qpOASES before, looks like seidel is much faster for toppra problem. As for the failed parameterization, I had a bug in my varying acceleration constraint which was causing ERR_UNKNOWN. ERR_FAIL_FORWARD_PASS was because of very short input trajectory.
